File: src/com/company/arrays/SecondLargest.java
package com.company.arrays;

public class SecondLargest {

    /**
     * Find Second Maximum Value in an Array
     */
    public static void main(String[] args) {

        int[] myArray = {1000,2,4,5,10,21,100};

        System.out.println(secondMaxNumber(myArray));

    }

    public static int secondMaxNumber(int[] array) {

        int largest = Integer.MIN_VALUE;
        int secondLargest = Integer.MIN_VALUE;

        for (int i = 0; i < array.length; i++) {
            if (array[i] > largest) {
                secondLargest = largest;
                largest = array[i];
            } else if (array[i] > secondLargest) {
                secondLargest = array[i];
            }
        }

        return secondLargest;
    }
}

File: src/com/company/arrays/SecondLargest.kt
package com.company.arrays

    /**
     * Find Second Maximum Value in an Array
     */
    fun main(){
        val myArray = intArrayOf(1000, 2, 4, 5, 10, 21, 100)

        println(secondMaxNumber(myArray))
    }

    fun secondMaxNumber(array: IntArray): Int {
        var largest = Int.MIN_VALUE
        var secondLargest = Int.MIN_VALUE

        for (i in array.indices) {
            if (array[i] > largest) {
                secondLargest = largest
                largest = array[i]
            } else if (array[i] > secondLargest) {
                secondLargest = array[i]
            }
        }

        return secondLargest
    }
